lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -804,13 +797,46 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
             # handle cluster change to main-orchestrator (i.e: init_hold: true -> false)
             self._handle_change_to_main_orchestrator_if_needed(event, previous_deployment_desc)
 
+        if not self.state.app.deployment_description:
+            logger.debug("Deployment description not yet computed, deferring event.")
+            event.defer()
+            return
+
         if self.upgrade_in_progress:
             logger.warning(
                 "Changing config during an upgrade is not supported. The charm may be in a broken, unrecoverable state"
             )
             event.defer()
             return
 
+        profile_restart_needed = False
+        try:
+            config_profile = self.profiles_manager.get_config_profile()
+            current_profile = self.state.unit.profile
+            self.status.clear(InvalidProfileConfigOption)
+        except ValueError:
+            logger.error(
+                "Invalid profile configuration. Value: %s", self.state.config.get("profile")
+            )
+            self.status.set(BlockedStatus(InvalidProfileConfigOption))
+            return
+
+        if self.profiles_manager.check_all_requirements(config_profile):
+            event.defer()
+            return
+
+        # if the profile hasn't been applied before
+        logger.debug("current profile: %s, config profile: %s", current_profile, config_profile)
+        if current_profile is None or current_profile != config_profile:
```

Comment:
I'm not sure I understand why we're checking if the current profile is none, doesn't this mean that it's a new deployment (where this hook will run before the _on_start)? in which case no restart needed